lib/zoo_user_warden_strategy.rb

```diff
@@ -1,12 +1,22 @@
-Warden::Strategies.add(:zoo_user) do
+require 'devise/strategies/authenticatable'
+
+class ZooUserWardenStrategy < Devise::Strategies::Authenticatable
   def valid?
-   params['user'] && params['user']['display_name'] && params['user']['password']
+    params['user'] && params['user']['display_name'] && params['user']['password']
   end
-
+  
   def authenticate!
     zoo_home_user = ZooniverseUser.authenticate(params['user']['display_name'],
                                                 params['user']['password'])
-    imported_user = zoo_home_user.import if zoo_home_user
-    imported_user ? success!(imported_user) : fail!("Could not log in")
+    if zoo_home_user && imported_user = zoo_home_user.import 
+      remember_me(imported_user)
+      imported_user.after_database_authentication
+      success!(imported_user)
+    end
+    
+    fail!("Invalid display_name or password.") unless zoo_home_user
```

Comment:
Yes I figured it didn't matter since we would delete all this code in about 2 weeks. Your PR has it so we should just resolve the merge in its favor.